### PR TITLE
Dhcpv4 cleanup

### LINKED ISSUE #8727: Network stack cleanup: DHCPv4
dhcpv4.c needs cleanup:
 . mostly declutter using #8723 rules
 . evaluate if k_work_delayed objects could be removed from struct net_if_dhcpv4 and made unique for each timer. See susbsy/net/l2/ethernet/arp.c for instance.

### PR DESCRIPTION
**Pull request by tbursztyka**
Fixes #8727

### CODE REVIEW

File: subsys/net/ip/dhcpv4.c [hunk 1]
```diff
@@ -262,12 +262,12 @@ static struct net_pkt *dhcpv4_prepare_message(struct net_if *iface, u8_t type,
 /* Prepare DHCPv4 Message request and send it to peer */
```
Comment (jukkar) [inline subsys/net/ip/dhcpv4.c]: Typo in "net/dhcpv4: Reorder ..." commit message. s/vaiable/variable/

File: subsys/net/ip/dhcpv4.c [hunk 2]
```diff
@@ -339,20 +339,11 @@ static void dhcpv4_send_request(struct net_if *iface)
...
-	} else {
-		ciaddr_buf = "0.0.0.0";
-	}
-
 	NET_DBG("send request dst=%s xid=0x%"PRIx32" ciaddr=%s"
 		"%s%s timeout=%"PRIu32"s",
 		net_sprint_ipv4_addr(server_addr),
 		iface->config.dhcpv4.xid,
-		ciaddr_buf,
+		ciaddr ? net_sprint_ipv4_addr(ciaddr) : "0.0.0.0",
```
Comment (jukkar) [inline subsys/net/ip/dhcpv4.c]: Nak to this. The net_sprint_ipv4_addr() uses static buffer internally so there can be only one call to this in one NET_DBG()

File: subsys/net/ip/dhcpv4.c [hunk 3]
```diff
@@ -396,130 +415,123 @@ static void dhcpv4_enter_selecting(struct net_if *iface)
...
 
-	/* Start rebinding time */
-	k_delayed_work_submit(&iface->config.dhcpv4.t2_timer,
-			      K_SECONDS(rebinding_time));
+	dhcpv4_update_timeout_work(iface->config.dhcpv4.request_time);
 }
 
-static void dhcpv4_timeout(struct k_work *work)
+#define IFACE_CONTAINER_OF(_dhcpv4)					\
+	CONTAINER_OF(CONTAINER_OF(_dhcpv4, struct net_if_config, dhcpv4), \
```
Comment (jukkar) [inline subsys/net/ip/dhcpv4.c]: This looks nasty. Why not the original way i.e., using just one CONTAINER_OF as that looks much cleaner?

**Review by jukkar — CHANGES_REQUESTED** [review]

File: subsys/net/ip/dhcpv4.c (hunk 3 above)
Comment (tbursztyka) [reply subsys/net/ip/dhcpv4.c]: here you can't use one container of: I need to get the net_if, but I have the net_if_dhcpv4, which is insinde net_if_config.

File: subsys/net/ip/dhcpv4.c (hunk 2 above)
Comment (tbursztyka) [reply subsys/net/ip/dhcpv4.c]: Indeed, I missed the fact it was already in use by server_addr.

**Review by jukkar — APPROVED** [review]
LGTM

**Pull request merged**